tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,463 @@
+				sc.WithHostpath(xfsHostpathDir),
+				sc.WithVolumeBindingMode("WaitForFirstConsumer"),
+				sc.WithReclaimPolicy("Delete"),
+				sc.WithParameters(map[string]string{
+					"enableXfsQuota": "true",
+					"softLimitGrace": "0%",
+					"hardLimitGrace": "0%",
+				}),
+			)
+			Expect(err).To(
+				BeNil(),
+				"while building StorageClass with name prefix {%s}",
+				scNamePrefix,
+			)
+
+			By("creating StorageClass API resource")
+			scObj, err = ops.SCClient.Create(context.TODO(), scObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating StorageClass with name prefix {%s}",
+				scNamePrefix,
+			)
+			scName = scObj.ObjectMeta.Name
+		})
+	})
+
+	When("pvc with storageclass "+scName+" is created", func() {
+		It("should create a pvc", func() {
+			By("building a PVC with StorageClass " + scName)
+			pvcObj, err = BuildPersistentVolumeClaim(pvcName, scName, capacity, accessModes)
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("creating above PVC with StorageClass " + scName)
+			_, err = ops.PVCClient.WithNamespace(namespaceObj.Name).Create(context.TODO(), pvcObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+		})
+	})
+
+	When("pod is created with pvc "+pvcName, func() {
+		It("should be up and running", func() {
+			By("building a pod with busybox image")
+			podObj, err = BuildPod(podName, pvcName, labelselector)
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("creating above pod with busybox image")
+			createdPod, err = ops.PodClient.WithNamespace(namespaceObj.Name).
+				Create(context.TODO(), podObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("verifying pod count as 1")
+			podCount := ops.GetPodRunningCountEventually(namespaceObj.Name, label, 1)
+			Expect(podCount).To(Equal(1), "while verifying pod count")
+		})
+	})
+
+	When("writing data more than quota limit into the hostpath volume", func() {
+		It("should not be able to write more than the enforced limit", func() {
+			By("Verifying the quota applied on the volume works")
+			execOption := NewOptions()
+			// The command formed below is attempting to create a file named “test.txt” inside the attached volume,
+			// its size increasing 1M in each iteration of the loop should show us the quota in action.
+			option := execOption.WithPodName(createdPod.Name).
+				WithContainer("busybox").
+				WithNamespace(namespaceObj.Name).
+				WithCommand([]string{"/bin/sh", "-c", fmt.Sprintf(`for i in $(seq 1 7); do dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=$i 2>/dev/null; if [ $? -ne 0 ]; then echo "filesize of $i Mb failed, quota reached!"; else echo "filesize of $i Mb was ok"; fi; done`)}...)
```

Comment:
Could you test if we are hitting the quota limit by using something like `du -h --summarize --block-size=M /mnt/store1` on the mountpoint?

Also why are we testing by writing in 1 MB increments, and not directly trying to exceed the quota limit? The error message returned by 'dd' could be useful if printed to make sure that the error was due to lack of capacity.

Can we do something like this...
```bash
dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=100 status=none || du -hs -BM /mnt/store1
```
This prints nothing if write succeeds, and does an exit 0, prints error and final size if write fails.
To add exit status value, you can do this also...
```bash
dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=100 status=none || (du -hs -BM /mnt/store1 && exit 1)
```

Reply:
> Also why are we testing by writing in 1 MB increments, and not directly trying to exceed the quota limit? The error message returned by 'dd' could be useful if printed to make sure that the error was due to lack of capacity.

It is required to test the behavior of feature... like if we write less data it should succeed once we start consuming over than limit it will throw error

Reply:
Check criteria might not be sufficient here.

With a check like `dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=100 status=none || du -hs -BM /mnt/store1`, whatever no. of bytes were written successfully, is printed --> proof of successful write.

Reply:
```suggestion
				WithCommand([]string{"/bin/sh", "-c", "dd if=/dev/zero of=/mnt/store1/test.txt bs=1M count=10 2>&1 || du -s -BM /mnt/outer | cut -f -1 1>&2"}...)
```

The above command can be used to try writing 10M. And the expected result is that the command should fail --> this being a -ve test.
The assertions after evaluating stdout and stderr can be something like this
```golang
Expect(stdOut).NotTo(BeEmpty(), "rying to write till the quota limit should be allowed")
Expect(stdErr).To(Equal(capacity), "trying to write beyond the quota limit should not be allowed")
```